iothub_client/samples/iothub_client_device_twin_and_methods_sample/iothub_client_device_twin_and_methods_sample.c

```diff
@@ -32,170 +42,550 @@
 
 #ifdef SAMPLE_MQTT
     #include "iothubtransportmqtt.h"
-#endif // SAMPLE_MQTT
-#ifdef SAMPLE_MQTT_OVER_WEBSOCKETS
+    static IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = MQTT_Protocol;
+#elif defined SAMPLE_MQTT_OVER_WEBSOCKETS
     #include "iothubtransportmqtt_websockets.h"
-#endif // SAMPLE_MQTT_OVER_WEBSOCKETS
-#ifdef SAMPLE_AMQP
+    static IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = MQTT_WebSocket_Protocol;
+#elif defined SAMPLE_AMQP
     #include "iothubtransportamqp.h"
-#endif // SAMPLE_AMQP
-#ifdef SAMPLE_AMQP_OVER_WEBSOCKETS
+    static IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = AMQP_Protocol;
+#elif defined SAMPLE_AMQP_OVER_WEBSOCKETS
     #include "iothubtransportamqp_websockets.h"
-#endif // SAMPLE_AMQP_OVER_WEBSOCKETS
-#ifdef SAMPLE_HTTP
+    static IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = AMQP_Protocol_over_WebSocketTls;
+#elif defined SAMPLE_HTTP
     #include "iothubtransporthttp.h"
-#endif // SAMPLE_HTTP
-
-#ifdef SET_TRUSTED_CERT_IN_SAMPLES
-#include "certs.h"
-#endif // SET_TRUSTED_CERT_IN_SAMPLES
-
-/* Paste in the your iothub device connection string  */
-static const char* connectionString = "[device connection string]";
-
-#define DOWORK_LOOP_NUM     3
+    static IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = HTTP_Protocol;
+#endif
 
-typedef struct MAKER_TAG
+// Connection String -- Paste in the iothub device connection string.
+static const char* connection_string = "[device connection string]";
+
+static IOTHUB_DEVICE_CLIENT_HANDLE iothub_client;
+static IOTHUB_TWIN_REQUEST_OPTIONS_HANDLE twin_request_options;
+
+static bool message_received = false;
+static int64_t twin_property_desired_version_value = 0;
+
+// Device Twin Properties
+#define TWIN_DESIRED "desired"
+#define TWIN_VERSION "$version"
+#define TWIN_MANUFACTURER "manufacturer"
```

Comment:
Humm TWIN_MANUFACTURER_PROPERTY_NAME, TWIN_MAKE_PROPERTY_NAME, etc?

Reply:
I hesitate to make these macro names longer because the code will get REALLY long when used.

Reply:
not that bad actually. updated.